lib/floe/workflow/choice_rule/data.rb

```diff
@@ -4,15 +4,20 @@ module Floe
   class Workflow
     class ChoiceRule
       class Data < Floe::Workflow::ChoiceRule
-        COMPARE_KEYS = %w[IsNull IsPresent IsNumeric IsString IsBoolean IsTimestamp String Numeric Boolean Timestamp].freeze
+        TYPES      = ["String", "Numeric", "Boolean", "Timestamp", "Present", "Null"].freeze
+        COMPARES   = ["Equals", "LessThan", "GreaterThan", "LessThanEquals", "GreaterThanEquals", "Matches"].freeze
+        # e.g.: (Is)(String), (Is)(Present)
+        TYPE_CHECK = /^(Is)(#{TYPES.join("|")})$/.freeze
+        # e.g.: (String)(LessThan)(Path), (Numeric)(GreaterThanEquals)()
+        OPERATION  = /^(#{(TYPES - %w[Null Present]).join("|")})(#{COMPARES.join("|")})(Path)?$/.freeze
 
         attr_reader :variable, :compare_key, :compare_predicate, :path
 
         def initialize(_workflow, _name, payload)
           super
 
           @variable = parse_path("Variable", payload)
-          parse_compare_key
+          parse_compare_key(payload)
```

Comment:
@kbrock isn't payload an ivar/attr_reader?  Do you need to pass it as a parameter or can you just let the attr_reader handle it?

```suggestion
          parse_compare_key
```

Reply:
Yes.

I want to drop `@payload` and it brings me small joys to avoiding using that instance variable.

See:
- https://github.com/ManageIQ/floe/pull/275

Reply:
Hm it is the static definition of whatever we have I'd be interested in why you want to drop it if it seems like we make use of it